v2/coord/conversion.go

```diff
@@ -2,61 +2,38 @@ package coord
 
 import (
 	"github.com/libp2p/go-libp2p/core/peer"
-	ma "github.com/multiformats/go-multiaddr"
 	"github.com/plprobelab/go-kademlia/kad"
 
 	"github.com/libp2p/go-libp2p-kad-dht/v2/kadt"
 )
 
-// NodeInfoToAddrInfo converts a kad.NodeInfo to a peer.AddrInfo.
-// This function will panic if info.ID() does not return a kadt.PeerID
-func NodeInfoToAddrInfo(info kad.NodeInfo[KadKey, ma.Multiaddr]) peer.AddrInfo {
-	peerID := info.ID().(kadt.PeerID)
-	return peer.AddrInfo{
-		ID:    peer.ID(peerID),
-		Addrs: info.Addresses(),
-	}
-}
-
-// NodeIDToAddrInfo converts a kad.NodeID to a peer.AddrInfo with no addresses.
+// KadPeerIDToAddrInfo converts a kad.NodeID to a peer.AddrInfo with no addresses.
 // This function will panic if id's underlying type is not kadt.PeerID
-func NodeIDToAddrInfo(id kad.NodeID[KadKey]) peer.AddrInfo {
+func KadPeerIDToAddrInfo(id kad.NodeID[KadKey]) peer.AddrInfo {
```

Comment:
still in use?

Reply:
I think so, but I can double check. I iiterated on checking whether I could remove these helpers after each round of refactoring.

Reply:
In use in a few places. Hopefully we can straighten out the types and have fewer conversion functions in the future